### DISCUSSION TITLE
[ARCHIVED] Откуда доставать типы для моделей?

### DISCUSSION THREAD

**Discussion by azinit** [feature-sliced/documentation#51] [Archive]
@ilyaagarkov поднял интересный вопрос 

*Вот есть у нас типы для моделей (`type User`, `type Post`, ...)*
*1. Должны ли мы их располагать в `entities`?*
   - *Можем ли мы тогда типы из одного `entity` импортировать в другом?*
 
*2. Или должны доставить из `shared/api`?*
   -  *А что если при кодогенерации (если она есть), нам не все модели подходят, и нужны свои?*
   -  *Создавать `src/models.ts`?*

---

> Думаю будет интересно послушать опыт - как каждый решал эту проблему

> UPD: Особенно, учитывая, что у многих в проектах кодогенерация, и модели генерируются в том числе "в одном месте" чаще всего, без разбивки по `entities` и т.д.

**Comment by ilyaagarkov** [comment]
Уточню - c типами данных приходящих из апи все понятно. Папка shared/api доступна для всех 
(хотя опять же может ли быть `api/` внутри entities?)

Но что делать а типами данных приходящих, например, из формы? Эти типы могут отличаться от типов из апи, а значит их где-то нужно хранить. Вопрос "где?", если может получится так что они нужны не в одной фиче.

**Reply by azinit** [reply]
> Уточню - c типами данных приходящих из апи все понятно. Папка shared/api доступна для всех

Соглы, особенно при условиях кодогенерации

**Reply by azinit** [reply]
> Но что делать а типами данных приходящих, например, из формы? Эти типы могут отличаться от типов из апи, а значит их где-то нужно хранить. Вопрос "где?", если может получится так что они нужны не в одной фиче.

Чаще всего да, я бы такие хранил локально с местом использования (например извне знать о типах с формы, о каком-то синтетическом типе данных - если он скорее всего только внутри и будет использоваться)
